src/Symfony/Component/Form/ChoiceList/ChoiceList.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\ChoiceList;
+
+use Symfony\Component\Form\ChoiceList\Factory\Cache\ChoiceFilter;
+use Symfony\Component\Form\ChoiceList\Factory\Cache\ChoiceFieldName;
+use Symfony\Component\Form\ChoiceList\Factory\Cache\ChoiceLabel;
+use Symfony\Component\Form\ChoiceList\Factory\Cache\ChoiceLoader;
+use Symfony\Component\Form\ChoiceList\Factory\Cache\ChoiceValue;
+use Symfony\Component\Form\ChoiceList\Factory\Cache\GroupBy;
+use Symfony\Component\Form\ChoiceList\Factory\Cache\PreferredChoice;
+use Symfony\Component\Form\ChoiceList\Loader\CallbackChoiceLoader;
+use Symfony\Component\Form\ChoiceList\Loader\ChoiceLoaderInterface;
+use Symfony\Component\Form\FormTypeInterface;
+
+/**
+ * A set of convenient static methods to create choice list callbacks or loaders.
+ *
+ * @author Jules Pietri <jules@heahprod.com>
+ */
+class ChoiceList
+{
+    /**
+     * @param FormTypeInterface     $formType A form type configuring a list
+     * @param ChoiceLoaderInterface $loader   Any pseudo callable to create a unique value from a choice
+     * @param mixed                 $vary     Dynamic data used to compute a unique hash when caching the loader
+     */
+    public static function lazy(FormTypeInterface $formType, callable $choices, $vary = null): ChoiceLoader
```

Comment:
Let's add a description for each method. Since this class is supposed to be used in user land, it should be really comprehensible.